emu/src/ppu.rs

```diff
@@ -1,2 +1,133 @@
 pub const DISPLAY_WIDTH: usize = 240;
 pub const DISPLAY_HEIGHT: usize = 160;
+pub const PALETTE_SIZE: usize = 16;
+
+pub const BG_PALETTE_ADDRESS: u32 = 0x05000000;
+pub const OBJ_PALETTE_ADDRESS: u32 = 0x05000200;
+
+use std::{cell::RefCell, fmt::Display, rc::Rc};
+
+#[cfg(feature = "debug")]
+use rand::Rng;
+
+use crate::{bitwise::Bits, internal_memory::InternalMemory};
+
+pub struct PixelProcessUnit {
+    internal_memory: Rc<RefCell<InternalMemory>>,
+}
+
+#[derive(PartialEq, Eq, Clone)]
+pub enum PaletteType {
+    BG,
+    OBJ,
+}
+
+pub struct PaletteColor {
+    pub red: u8,
+    pub green: u8,
+    pub blue: u8,
+}
+
+impl Display for PaletteColor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "({},{},{})", self.red, self.green, self.blue)
+    }
+}
+
+impl PixelProcessUnit {
+    pub fn new(internal_memory: Rc<RefCell<InternalMemory>>) -> Self {
+        Self { internal_memory }
+    }
+
+    pub fn get_bg_palette_ram(&self) -> [u8; 0x200] {
+        self.internal_memory.borrow_mut().bg_palette_ram
+    }
+
+    pub fn get_obj_palette_ram(&self) -> [u8; 0x200] {
+        self.internal_memory.borrow_mut().obj_palette_ram
+    }
+
+    pub fn color_from_two_u8(&self, upper_byte: u8, lower_byte: u8) -> PaletteColor {
+        let mut upper: u32 = upper_byte.into();
```

Comment:
I have some doubts about this: is it better to implement `Bits` trait also for u8?